File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        ones=s.count("1")
        zero,ans=0,0
        
        for i in range(len(s)-1):
            if(s[i]=="1"):
                ones -=1
            else:
                zero+=1
            ans=max(ans,ones+zero)

        return ans
        

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int n = s.length();
        int oneCount = 0, zeroCount = 0;

        for (int i = 0; i < n; i++) {
            if (s.charAt(i) == '1')
                oneCount++;
        }
        int ans = 0;
        for (int i = 0; i < n - 1; i++) {
            if (s.charAt(i) == '1') {
                oneCount--;
            } else {
                zeroCount++;
            }
            ans = Math.max(ans, (oneCount + zeroCount));
        }
        return ans;

    }
}